src/lib.rs

```diff
@@ -166,16 +169,83 @@ pub enum PixelLayout {
     Unknown,
 }
 
+#[derive(Copy, Clone, Debug)]
+pub enum PlanarImageComponent {
+    Y,
+    U,
+    V,
+}
+
+impl std::convert::From<usize> for PlanarImageComponent {
+    fn from(index: usize) -> Self {
+        match index {
+            0 => PlanarImageComponent::Y,
+            1 => PlanarImageComponent::U,
+            2 => PlanarImageComponent::V,
+            _ => panic!("Invalid YUV index: {}", index),
+        }
+    }
+}
+
+impl std::convert::From<PlanarImageComponent> for usize {
+    fn from(component: PlanarImageComponent) -> Self {
+        match component {
+            PlanarImageComponent::Y => 0,
+            PlanarImageComponent::U => 1,
+            PlanarImageComponent::V => 2,
+        }
+    }
+}
+
+pub struct Plane(Picture, PlanarImageComponent);
+impl AsRef<[u8]> for Plane {
+    fn as_ref(&self) -> &[u8] {
+        unsafe {
+            std::slice::from_raw_parts(
+                self.0.plane_data_ptr(self.1) as *const u8,
+                self.0.width() as usize * self.0.height() as usize,
```

Comment:
I don't think the size is correct for the U/V planes in case of subsampling

Reply:
I.e. you have to work here with a) the stride for each plane and b) the plane height for the specific plane